### DISCUSSION TITLE
Server side rendering (SSR)

### DISCUSSION THREAD

**Discussion by Immortalin** [wasp-lang/wasp#927] [Archived]
**Is your feature request related to a problem? Please describe.**
Pre render react UI server side before sending to client.

**Describe the solution you'd like**
Add Next.js as a target

**Comment by Martinsos** [comment]
Makes sense! I am somewhat reluctant about making Next our target though, since it is (I think?) not as flexible as an Express app, but I am not ruling it out as an option completely. What if we wanted to have SSR in ExpressJS app? Is there an easy way to go about that?

**Comment by samos123** [comment]
I would need server side rendering for SEO. It should be relatively easy to enable SSR out of the box like Next.js while still providing flexibility.

**Comment by Martinsos** [comment]
@joanreyero also asked if we could support SSR in the future, for SEO.

**Comment by Martinsos** [comment]
@Fecony also requested this.

**Comment by Martinsos** [comment]
More feedback here: https://www.reddit.com/r/javascript/comments/z7xo9t/comment/iyagm7a/?utm_source=share&utm_medium=web2x&context=3 .

Very shrotly, reddit user novagenesis says that things he would like from NextJS are:
1. almost-seamless hydration
2. being able to initial-render with serverside data

**Comment by Martinsos** [comment]
Another reddit user mentioned it might be interesting doing server side templating based on TSX.
https://www.reddit.com/r/javascript/comments/z7xo9t/comment/iyc2ahq/?utm_source=share&utm_medium=web2x&context=3
